gigascience/gigadb-website
# Provide horizontal scroll bar in files table

# Pull request for issue: #1596

This is a pull request for the following functionalities:

* Display horizontal scroll bar in dataset files table.
* Provide Playwright test to check for the presence of horizontal scroll bar.
* Enable Playwright test to be executed in dev environment with a Docker container.

## How to test?

Deploy a local GigaDB web application:
```
$ cd /path/to/gigadb-website
$ ./up.sh
```

Go to http://gigadb.gigasciencejournal.com:9170/dataset/102484 in your browser and confirm the presence and functionality of a horizontal scroll bar in this dataset's files table. Interestingly, the scroll bar is only required and displayed in Chrome and Safari browsers. Firefox can display the whole dataset files table without using a scroll bar. This issue #1596 was reported in Chrome browser.

Run playwright test:
```
$ docker-compose run --rm playwright npm run test dataset-table-scrollbars.spec.js
```

## How have functionalities been implemented?

The fix described [here](http://sforsuresh.in/datatables-header-body-not-aligned/) was implemented by wrapping a div around the DataTable with style "overflow as auto" using a CSS class called `dataset-datatables-wrapper` which can be found in `dataset.less`.

## Any issues with implementation?

Originally, [scrollX](https://datatables.net/reference/option/scrollX) was tested to provide the horizontal scroll bar in the dataset file table but it causes the mis-alignment of the column header with the rows in the table. 

## Any changes to automated tests?

A new Playwright test in dataset-table-scrollbars.spec.js has been created to test for the presence of a horizontal scroll bar in dataset files table. This test can be executed using a Docker container. Previously, running Playwright tests involved installing Node on your development machine.

The new Playwright test required providing dev data with a new dataset 102484 that contains data requiring a horizontal scroll to view all its data.

## Any changes to documentation?

The README.md for Playwright tests has been expanded with information on how to run the new Playwright test.
